packages/app/src/settings/project/renderers/RenderPrimitive.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <Tooltip
+    class="inline"
+    placement="right"
+  >
+    <span
+      :class="props.class"
+      :data-cy-config="props.from"
+    >
+      {{ content }},
+    </span>
+    <template #popper>
+      {{ props.from }}
+    </template>
+  </Tooltip>
+</template>
+
+<script lang="ts" setup>
+import Tooltip from '@packages/frontend-shared/src/components/Tooltip.vue'
+import { computed } from 'vue'
+
+const props = defineProps<{
+  value?: string | number| null
+  from: string
+  class?: string
+}>()
+
+const content = computed(() => {
+  if (props.value === null || props.value === undefined) {
+    return 'null'
+  }
+
+  if (typeof props.value === 'string') {
+    if (props.value.startsWith('[Function')) {
```

Comment:
Do we have tests for all these different things? Might be good to add a `RendererPrimitive.cy.tsx` spec and use `cy.realHover` to verify the content is actually correct for each branch here.